### ISSUE TITLE
let declaration in a for..of loop results in broken code

### ISSUE THREAD

**Issue by tec27** [babel/babel#1169]
With the code:
```javascript
function foo() {
  let input = ['a', 'b', 'c']
  let output = {}

  for (let c of input) {
    let name = c
    output[name] = name
  }

  return output
}
```

The output from babel is:
```javascript
'use strict';

function foo() {
  var input = ['a', 'b', 'c'];
  var output = {};

  var _iteratorNormalCompletion = true;
  var _didIteratorError = false;
  var _iteratorError = undefined;

  try {
    for (var _iterator = input[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = true) {
      var c = _step.value;

      var _name = c;
      output[name] = name;
    }
  } catch (err) {
    _didIteratorError = true;
    _iteratorError = err;
  } finally {
    try {
      if (!_iteratorNormalCompletion && _iterator['return']) {
        _iterator['return']();
      }
    } finally {
      if (_didIteratorError) {
        throw _iteratorError;
      }
    }
  }

  return output;
}
```

Specifically, the problematic generated code is here:
```javascript
var _name = c;
output[name] = name;
```

It should be `output[_name] = _name;`

This works fine in a regular for loop, and works fine if `name` is declared with `var`

**Comment by muzuiget** [comment]
Same here, any workaround? I stuck at this bug from traceur to babel.

**Comment by hansl** [comment]
Workaround is to not name your variables the same name within the for loop, but that defeats the purpose of a `let` in the first place :(

**Comment by sebmck** [comment]
Sorry for the delay, fixed as of 5.0.11. Thanks everyone :smile:
